bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirSerialConnector.java

```diff
@@ -0,0 +1,635 @@
+
+                        // check for start and end sequence and if the response cmd
+                        // matches
+                        // 11 is the minimum response length with one data byte
+                        if (responseBlock.length >= 11 && responseBlock[2] == (byte) 0x07
+                                && responseBlock[3] == (byte) 0xf0
+                                && responseBlock[responseBlock.length - 2] == (byte) 0x07
+                                && responseBlock[responseBlock.length - 1] == (byte) 0x0f
+                                && (responseBlock[5] & 0xff) == command.getReplyCmd()) {
+
+                            if (logger.isTraceEnabled()) {
+                                logger.trace("receive RAW DATA: {}", dumpData(responseBlock));
+                            }
+
+                            byte[] cleanedBlock = cleanupBlock(responseBlock);
+
+                            int dataSize = cleanedBlock[2];
+
+                            // the cleanedBlock size should equal dataSize + 2 cmd
+                            // bytes and + 1 checksum byte
+                            if (dataSize + 3 == cleanedBlock.length - 1) {
+
+                                byte checksum = cleanedBlock[dataSize + 3];
+                                int[] replyData = new int[dataSize];
+                                for (int i = 0; i < dataSize; i++) {
+                                    replyData[i] = cleanedBlock[i + 3] & 0xff;
+                                }
+
+                                byte[] _block = new byte[3 + replyData.length];
+                                System.arraycopy(cleanedBlock, 0, _block, 0, _block.length);
+
+                                // validate calculated checksum against submitted
+                                // checksum
+                                if (calculateChecksum(_block) == checksum) {
+                                    if (logger.isTraceEnabled()) {
+                                        logger.trace("receive CMD: {} DATA: {}",
+                                                String.format("%02x", command.getReplyCmd()), dumpData(replyData));
+                                    }
+                                    send(ACK);
+
+                                    return replyData;
+                                }
+
+                                logger.debug("Unable to handle data. Checksum verification failed");
+                            } else {
+                                logger.debug("Unable to handle data. Data size not valid");
+                            }
+
+                            if (logger.isTraceEnabled()) {
+                                logger.trace("skip CMD: {} DATA: {}", String.format("%02x", command.getReplyCmd()),
+                                        dumpData(cleanedBlock));
+                            }
+                        }
+                    }
+                } catch (IOException e) {
```

Comment:
You might catch an `InterruptedIOException` here, so you want to make sure that interrupt it handled properly.

Reply:
Not sure if interrupting the thread is enough or I also have to throw a RuntimeException or somthing. Pls check

Reply:
Not sure if interrupting the thread is enough or I also have to throw a RuntimeException or somthing. Pls check